src/main/java/org/jboss/set/pull/processor/impl/evaluator/PullRequestVsStreamComponentLabelEvaluator.java

```diff
@@ -0,0 +1,69 @@
+package org.jboss.set.pull.processor.impl.evaluator;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+
+import org.jboss.set.pull.processor.EvaluatorContext;
+import org.jboss.set.pull.processor.ProcessorPhase;
+import org.jboss.set.pull.processor.data.DefinedLabelItem;
+import org.jboss.set.pull.processor.data.EvaluatorData;
+import org.jboss.set.pull.processor.data.LabelData;
+import org.jboss.set.pull.processor.data.LabelItem;
+import org.jboss.set.pull.processor.data.PullRequestData;
+
+/**
+ * Check if PR repository and branch match those defined in aphrodite.
+ *
+ * @author baranowb
+ *
+ */
+public class PullRequestVsStreamComponentLabelEvaluator extends AbstractLabelEvaluator {
+
+    @Override
+    public void eval(EvaluatorContext context, EvaluatorData data) {
+        final LabelData labelData = super.getLabelData(EvaluatorData.Attributes.LABELS_CURRENT, data);
+        final PullRequestData pullRequestData = data.getAttributeValue(EvaluatorData.Attributes.PULL_REQUEST_CURRENT);
+        final LabelData labelDataUpstream = super.getLabelData(EvaluatorData.Attributes.LABELS_UPSTREAM, data);
+        final PullRequestData pullRequestDataUpstream = data.getAttributeValue(EvaluatorData.Attributes.PULL_REQUEST_UPSTREAM);
+        scrutinize(labelData, pullRequestData);
+        scrutinize(labelDataUpstream, pullRequestDataUpstream);
+    }
+
+    protected void scrutinize(final LabelData labelData, final PullRequestData pullRequestData) {
+        try {
+            //this might happen in case of upstream either not required or
+            if (!pullRequestData.isRequired() || pullRequestData.getPullRequest() == null) {
```

Comment:
> either not required or ??

I don't follow the first part `!pullRequestData.isRequired()` in the IF. 

It's always required as removal in LinkedIssuesEvaluator makes no chance to set upstreamIssueData to `notRequired`
```
            // TODO: XXX move those to PullRequest ?
            if (upstreamIssue == null && context.getPullRequest().isUpstreamRequired()) {
                upstreamIssueData.notRequired();
            }
```
Due to this, it won't set upstreamPullRequestData to `notRequired()` in LinkedPullRequestEvaluator. 
```
            if (!upstreamIssue.isRequired())
                upstreamPullRequestData.notRequired();
```
I think last major re-factoring removed detection of keyword like "**no upstream required**" which defined legitimate not-required.

Reply:
My bad it seems, thought I got it right, revoked removal of that code.